### ISSUE TITLE
[BUG] Race condition in AddressFinder

### ISSUE THREAD

**Issue by DuckThom** [tig-nl/postnl-magento2#250]
**To Reproduce**

It's not easy to reproduce and it's possible that this will occur in more "mature" Magento installations (heavier load on the checkout)

1. Go to the Magento checkout
2. Make sure a non-NL (ie. Belgian) address is pre-selected from the address book and that the quote already has a non-NL address attached to it
3. Check if the request data for the `postnl/deliveryoptions/save` request matches the pre-selected address

**Expected result**
The product code in the tig_postnl_order table should not be the default 3085 (domestic NL package) but instead be for the selected country (ie. in the 4000 range if a Belgian address was used) and the request data for the `postnl/deliveryoptions/save` request should match the pre-selected address.
Since the product code is now for a domestic NL package but the quote address is a Belgian address, an error is generated when a packing slip label is requested for the shipment due to an incorrect product code for the PostNL shipment: "Countries in receiver and sender address not allowed together for this product"

**Actual result**
The uiRegistry request resolves after the quote address was returned in the AddressFinder causing it to overwrite/modify the already returned data that is being processed by the other logic. As seen in the screenshot below, the logged address has the correct quote information, however halfway through the `Delivery::getDeliverydays` flow, the data was changed by the uiRegistry callback.

The call `uiRegistry.get(RegistryFields, function(....` dispatches a request (not HTTP, but something like a Promise) internally, which will resolve once all the requested fields are present. Once it resolves, the callback is executed. However, since it's resolving _after_ the quote address data was put in the `address` variable and that data was returned, it's overwriting that data since it is all using the same object pointer that's in the `address` variable in the AddressFinder.

**Workaround**
Write the quote address data to a different/new variable in the AddressFinder so that the uiRegistry callback does not overwrite the existing data.

https://github.com/tig-nl/postnl-magento2/blob/c5570db63943834b1dbb567cd9e1e2184ea93a5a/view/frontend/web/js/Helper/AddressFinder.js#L68

```
-           address = {
+           var tempaddress = {
                street: shippingAddress.street[0],
                postcode: shippingAddress.postcode,
                housenumber: housenumber,
                firstname: shippingAddress.firstname,
                lastname: shippingAddress.lastname,
                telephone: shippingAddress.telephone,
                country: shippingAddress.countryId
            };

-           if (address.country && address.postcode && address.street !== undefined && address.street[0] && address.housenumber) {
-               return address;
+           if (tempaddress.country && tempaddress.postcode && tempaddress.street !== undefined && tempaddress.street[0] && tempaddress.housenumber) {
+               return tempaddress;
            }
```

**Screenshots**

![image](https://user-images.githubusercontent.com/6512901/118788886-84589300-b894-11eb-8eaf-fd8a62d1f317.png)

The address data changes halfway through the flow of the address/delivery moment processing.

(Note: image does not contain private information, just a randomly selected address used for testing)

- Magento version: 2.4.2
- PHP version: 7.4.15
- TIG PostNL version: 1.9.8

**Comment by tig-jeffreybranderhorst** [comment]
Hi @DuckThom ,

Thank you for submitting this and for the detailed explanation.  We are going to investigate this further thank you!

Have a great day,
Jeffrey

**Comment by tig-vincentthart** [comment]
Hi @DuckThom ,

This has been resolved in the latest release (v1.9.9). If you ran into any issues, feel free to contact the PostNL support.

Greetings,

Vincent
